Source/WebKit/UIProcess/mac/TextCheckerMac.mm

```diff
@@ -84,12 +84,23 @@ static bool shouldAutomaticDashSubstitutionBeEnabled()
     return [defaults boolForKey:WebAutomaticDashSubstitutionEnabled];
 }
 
+static bool shouldGrammarCheckingBeEnabled()
+{
+    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+#if USE(NSSPELLCHECKER_GRAMMAR_CHECKING_POLICY)
+    if (![defaults objectForKey:WebGrammarCheckingEnabled])
```

Comment:
Nit - you can probably cache the value here to avoid asking for it twice.

Reply:
Will do, was just following the structure of the other defaults above.

Reply:
Actually, we might need to keep as is since `boolForKey` handles both `NSNumber` and `NSString`.